src/main/ruby/core/io.rb

```diff
@@ -2631,7 +2631,7 @@ def write(data)
     end
 
     if @sync
-      Truffle::POSIX.write_string @descriptor, data
+      Truffle::POSIX.write_string_with_retry self, data
```

Comment:
Shouldn't IO#write retry?

Reply:
Yes, which is why it calls `write_string_with_retry` and not `write_string_no_retry`.